osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/endpoints/PublicLightingAdHocManagementEndpoint.java

```diff
@@ -94,14 +98,34 @@ public FindAllDevicesResponse findAllDevices(@OrganisationIdentification final S
         final FindAllDevicesResponse response = new FindAllDevicesResponse();
 
         try {
+            PageSpecifier pageSpecifier;
+            if (request.getPageSize() == null) {
+                pageSpecifier = new PageSpecifier(PAGE_SIZE, request.getPage());
+            } else {
+                pageSpecifier = new PageSpecifier(request.getPageSize(), request.getPage());
+            }
             final Page<Device> page = this.adHocManagementService.findAllDevices(organisationIdentification,
-                    request.getPage());
-
-            final DevicePage devicePage = new DevicePage();
-            devicePage.setTotalPages(page.getTotalPages());
-            devicePage.getDevices().addAll(this.adHocManagementMapper.mapAsList(page.getContent(),
-                    org.opensmartgridplatform.adapter.ws.schema.publiclighting.adhocmanagement.Device.class));
-            response.setDevicePage(devicePage);
+                    pageSpecifier);
+
+            if (page != null) {
```

Comment:
Fixed

Reply:
I took a better look at the schema definition (pl-adhocmanagement.xsd) this time.
It looks like the `DevicePage` is a required element of the `FindAllDevicesResponse`.
I will look into it a bit further, and probably change the code myself to add some kind of empty `DevicePage` if no devices are returned by the `adHocManagementService`.
Please let me know if you disagree, then we could talk about this a bit more,

Reply:
Feel free to update it.

Reply:
I updated the license headers and made a few changes to the code and XSD in case there are no devices.
The request without the devices turned out to be invalid with regards to the XSD as I suspected.
Since I don't think I can (or should?) push to your branch, I branched of the tip of your changes and created pull request #534.
I asked someone else to take a look at my changes, and when they are found OK, your additions will be merged with the other pull request.